daemon/api.go

```diff
@@ -627,6 +628,9 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
 		if q != "" {
 			return BadRequest("cannot use 'q' and 'name' together")
 		}
+		if commonID != "" {
+			return BadRequest("cannot use 'common-id' and 'name' together")
+		}
```

Comment:
we don't have tests for the error cases? should we?

Reply:
we didn't; i've added a test for all bad combos